fix(mcp): retain RagMcpAgent stub class to satisfy migration history

## 目的

#109 の merge 後に `wrangler deploy` を走らせたところ下記エラーで失敗。旧 `RagMcpAgent` class を stub として残して解消する。

```
[ERROR] New version of script does not export class 'RagMcpAgent' which is
depended on by existing Durable Objects. Did you forget to include it?
If you renamed it, try a rename-class migration. [code: 10064]
```

## 前提 (検証済み)

- #109 で class `RagMcpAgent` → `RagMcpAgentV2` に rename、migration v2 で `new_sqlite_classes=[RagMcpAgentV2]` 追加
- Cloudflare は v1 の `new_sqlite_classes=[RagMcpAgent]` で declare したクラスを script が export し続けることを要求する
- rename-class migration を後付けで追加する選択肢もあるが、renamed_classes は state を保持するため in-memory init() cache も引き継がれる可能性が残る。別 namespace を作って fresh init() を強制したい本件の目的に反する

## 制約

- V2 は別 DO namespace として新 init() を走らせたい (本 PR 目的)
- 旧 RagMcpAgent storage はこの PR では触らず orphan のまま残す (別 PR で deleted_classes 経由で掃除)
- stub class は traffic を受けないため、最小の DurableObject 実装で OK

## 対象ファイル

- `src/mcp.ts` — `RagMcpAgent` を `DurableObject` 継承の空 stub として残す
- `src/index.ts` — stub の export を追加

## 関連

- #109 (force reset RagMcpAgent DO init() schema cache)
- #110 (直前の merge PR)

--- a/src/index.ts
+++ b/src/index.ts
@@ -38,6 +38,9 @@ import { RagMcpAgentV2 } from "./mcp.js";
 // Durable Object: issue/PR state store (SQLite-backed)
 export { IssueStore } from "./store.js";
 
+// Durable Object: MCP server — legacy stub (retained for migration compatibility only).
+export { RagMcpAgent } from "./mcp.js";
+
 // Durable Object: MCP server (tools: search_issues, get_issue_context, list_recent_activity)
 export { RagMcpAgentV2 } from "./mcp.js";
 

--- a/src/mcp.ts
+++ b/src/mcp.ts
@@ -12,6 +12,7 @@
  * idFromName("user-{githubUserId}").
  */
 
+import { DurableObject } from "cloudflare:workers";
 import { McpAgent } from "agents/mcp";
 import { McpServer } from "@modelcontextprotocol/sdk/server/mcp.js";
 import { z } from "zod";
@@ -64,6 +65,20 @@ function githubHeaders(token: string): Record<string, string> {
  * additional queries rather than lifting this cap.
  */
 const INCLUDE_CONTENT_MAX_DOCS = 5;
+
+/**
+ * Legacy class retained solely to satisfy Cloudflare's "class must exist in
+ * script for classes declared in past migrations (v1 new_sqlite_classes)"
+ * constraint. MCP_OBJECT binding points to RagMcpAgentV2 now, so this class
+ * never receives live traffic. It exists only so `wrangler deploy` does not
+ * fail with "script does not export class 'RagMcpAgent'". A follow-up PR can
+ * delete this class together with a deleted_classes migration.
+ */
+export class RagMcpAgent extends DurableObject<Env> {
+  async fetch(): Promise<Response> {
+    return new Response("RagMcpAgent has been retired; use RagMcpAgentV2", { status: 410 });
+  }
+}
 
 export class RagMcpAgentV2 extends McpAgent<Env, unknown, McpProps> {
   // @ts-expect-error -- McpServer version mismatch between top-level SDK and agents' bundled copy (same issue as webhook-mcp; wrangler resolves at bundle time)

